src/converters/investimentalConverter.ts

```diff
@@ -0,0 +1,212 @@
+            // Populate the progress bar.
+            const bar1 = this.progress.create(orderGroups.size, 0);
+
+            for (const [orderID, orderRecords] of orderGroups) {
+                const combinedRecord = this.combineRecords(orderRecords);
+
+                if (combinedRecord) {
+
+                    const currency = this.extractCurrency(combinedRecord.accountName);
+
+                    if (!currency) {
+                        this.progress.log(`[i] Could not determine currency for order ${orderID}. Skipping...\n`);
+                        bar1.increment();
+                        continue;
+                    }
+                
+                    let security: YahooFinanceRecord;
+                    try {
+                        security = await this.securityService.getSecurity(
+                            null,
+                            combinedRecord.symbol,
+                            null,
+                            currency,
+                            this.progress);
+                    }
+                    catch (err) {
+                        return errorCallback(err);
+                    }
+
+                    if (!security) {
+                        this.progress.log(`[i] No result found for action ${combinedRecord.side}, symbol ${combinedRecord.symbol}, currency ${currency}! Please add this manually..\n`);
+                        bar1.increment();
+                        continue;
+                    }
+
+                    const activity = this.createActivity(combinedRecord, security);
+                    if (activity) {
+                        result.activities.push(activity);
+                    }
+                }
+
+                bar1.increment();
+            }
+
+        this.progress.stop()
+
+        successCallback(result);
+        });
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public isIgnoredRecord(record: InvestimentalRecord): boolean {
+        // Ignore Out Of Time records
+        return record.updateType === "OOT";
+    }
+
+    private groupRecordsByorderID(records: InvestimentalRecord[]): Map<string, InvestimentalRecord[]> {
+        const orderGroups = new Map<string, InvestimentalRecord[]>();
+
+        for (const record of records) {
+            if (this.isIgnoredRecord(record)) {
+                continue;
+            }
+            if (!orderGroups.has(record.orderID)) {
+                orderGroups.set(record.orderID, []);
+            }
+            orderGroups.get(record.orderID)!.push(record);
+        }
+
+        return orderGroups;
+    }
+
+    private combineRecords(orderRecords: InvestimentalRecord[]): InvestimentalRecord | null {
+        orderRecords.sort((a, b) => new Date(a.updateTime).getTime() - new Date(b.updateTime).getTime());
+
+        let remainingVolume = 0;
+        let executedVolume = 0;
+        let totalValue = 0;
+        let lastFee = 0;
+        let lastPrice = 0;
+    
+        for (const record of orderRecords) {
```

Comment:
I think I understand what happens here, in relation to the input file.

Do I read it correctly that the input file contains a lot (a lot) of small partial order fills and then you convert them all together in one large order?

Reply:
This is partly correct.

Indeed there are small partial order fils (see Order ID 13268) and I decided there is no point in having this level of granularity so I convert them in one order because this reflects my in input order. I'm not really interested in how many pieces the broker executes my order.

But the majority of the records are with Update Type `Chg` because I've updated these orders many times before they were executed.